src/lib/vesting.ts

```diff
@@ -37,50 +47,113 @@ export async function withdrawVested(
   );
   const signer = wallet.signer;
 
-  state.set("withdrawingSign");
+  state.set({ type: "withdrawingSign" });
 
-  const tx = await contract.connect(signer).withdrawVested();
+  try {
+    const tx: TransactionResponse = await contract
+      .connect(signer)
+      .withdrawVested();
 
-  state.set("withdrawing");
-  await tx.wait();
+    state.set({ type: "withdrawing", tx });
+    await tx.wait();
+  } catch (e) {
+    handleEtherErrorState(e, "Unknown error, check the dev console");
+    return;
+  }
   session.state.refreshBalance(wallet);
-  state.set("withdrawn");
+  state.set({ type: "withdrawn" });
 }
 
-export async function getInfo(
-  address: string,
-  wallet: Wallet,
-): Promise<VestingInfo> {
-  const contract = new ethers.Contract(
-    address,
-    ethereumContractAbis.vesting,
-    wallet.provider,
-  );
-  const token = await contract.token();
-  const beneficiary = await contract.beneficiary();
-  const withdrawable = await contract.withdrawableBalance();
-  const withdrawn = await contract.withdrawn();
-  const total = await contract.totalVestingAmount();
-  const vestingStartTime = await contract.vestingStartTime();
-  const vestingPeriod = await contract.vestingPeriod();
-  const cliffPeriod = await contract.cliffPeriod();
-
-  const tokenContract = new ethers.Contract(
-    token,
-    ethereumContractAbis.token,
-    wallet.provider,
-  );
-  const symbol = await tokenContract.symbol();
-
-  return {
-    token,
-    symbol,
-    beneficiary,
-    totalVesting: utils.formatBalance(total),
-    withdrawableBalance: utils.formatBalance(withdrawable),
-    withdrawn: utils.formatBalance(withdrawn),
-    vestingStartTime,
-    vestingPeriod,
-    cliffPeriod,
-  };
+export const getInfo = cache.cached(
+  async (address: string, wallet: Wallet): Promise<VestingInfo | undefined> => {
+    const contract = getVestingContract(address, wallet);
+
+    let vestingInfo:
+      | [
+          string,
+          string,
+          BigNumber,
+          BigNumber,
+          BigNumber,
+          string,
+          string,
+          string,
+        ]
+      | undefined = undefined;
+    let token: string | undefined = undefined;
+
+    try {
+      token = await contract.token();
+      if (!token) {
+        return undefined;
+      }
+
+      const tokenContract = new ethers.Contract(
+        token,
+        ethereumContractAbis.token,
+        wallet.provider,
+      );
+
+      vestingInfo = await Promise.all([
+        tokenContract.symbol(),
+        contract.beneficiary(),
+        contract.withdrawableBalance(),
+        contract.withdrawn(),
+        contract.totalVestingAmount(),
+        contract.vestingStartTime(),
+        contract.vestingPeriod(),
+        contract.cliffPeriod(),
+      ]);
+    } catch (e) {
+      return undefined;
```

Comment:
Can we log this error with `console.warn`?

Reply:
16ead15